client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx

```diff
@@ -0,0 +1,360 @@
+				{
+					components: { learnMoreLink: learnMoreLink( MAP_EXISTING_DOMAIN ) },
+					args: { domainName: domain.name },
+				}
+			);
+		}
+
+		return (
+			<React.Fragment>
+				<div>
+					<p>{ primaryMessage }</p>
+					{ ! isSubdomain( domain.name ) && (
+						<ul>
+							{ WPCOM_DEFAULTS.map( nameServer => {
+								return <li key={ nameServer }>{ nameServer }</li>;
+							} ) }
+						</ul>
+					) }
+				</div>
+				<div>{ secondaryMessage }</div>
+			</React.Fragment>
+		);
+	}
+
+	renderDefaultRenewButton() {
+		const { domain, translate } = this.props;
+
+		if ( domain.expired || isExpiringSoon( domain, 30 ) ) {
+			return null;
+		}
+
+		if ( domain.bundledPlanSubscriptionId ) {
+			return (
+				<div>
+					<RenewButton
+						compact={ true }
+						selectedSite={ this.props.selectedSite }
+						subscriptionId={ parseInt( domain.bundledPlanSubscriptionId, 10 ) }
+						customLabel={ translate( 'Renew your plan' ) }
+						tracksProps={ { source: 'mapped-domain-status', mapping_status: 'active-plan' } }
+					/>
+				</div>
+			);
+		}
+
+		return (
+			<div>
+				<RenewButton
+					compact={ true }
+					selectedSite={ this.props.selectedSite }
+					subscriptionId={ parseInt( domain.subscriptionId, 10 ) }
+					tracksProps={ { source: 'mapped-domain-status', mapping_status: 'active' } }
+				/>
+			</div>
+		);
+	}
+
+	renderAutoRenew() {
+		return (
+			<Card compact={ true }>
+				Auto Renew (on) <CompactFormToggle checked={ true } />
+			</Card>
+		);
+	}
+
+	handlePaymentSettingsClick = () => {
+		this.props.recordPaymentSettingsClick( this.props.domain );
+	};
+
+	render() {
+		const { domain, moment, translate } = this.props;
+		const { name: domain_name } = domain;
+
+		const { statusText, statusClass, icon } = this.resolveStatus();
+
+		const newStatusDesignAutoRenew = config.isEnabled( 'domains/new-status-design/auto-renew' );
+
+		return (
+			<div className="domain-types__container">
+				<DomainStatus
+					header={ domain_name }
+					statusText={ statusText }
+					statusClass={ statusClass }
+					icon={ icon }
+				>
+					{ this.renderSettingUpNameservers() }
+					{ this.renderExpiringSoon() }
+				</DomainStatus>
+				<Card compact={ true } className="domain-types__expiration-row">
+					<div>
+						{ domain.bundledPlanSubscriptionId
+							? translate( 'Expires with your plan on %(expiry_date)s', {
+									args: {
+										expiry_date: moment( domain.expiry ).format( 'LL' ),
+									},
+							  } )
+							: translate( 'Expires: %(expiry_date)s', {
+									args: {
+										expiry_date: moment( domain.expiry ).format( 'LL' ),
+									},
+							  } ) }
+					</div>
+					{ this.renderDefaultRenewButton() }
+					{ ! newStatusDesignAutoRenew && (
+						<div>
+							<SubscriptionSettings
+								type={ domain.type }
+								compact={ true }
+								subscriptionId={ domain.subscriptionId }
```

Comment:
Should this go to the plan instead if the mapping is bundled with the plan? Then that's where they would manage the payment settings. Though, I understand that the payments page is useful for bundled mappings because that's where you can actually remove them from the site... though, the user might not now that 😅 Up to you both, @hambai and @fditrapani, just wanted to flag this, maybe it was already discussed.

Reply:
The idea is to remove the Payment Setting button once we implement the autorenew toggle and the Delete menu. Till then just to be safe we shoul keep the button and I think it makes sense for it to point to the actual mapping purchase page.